spock-core/src/main/java/org/spockframework/runtime/extension/builtin/RollupExtension.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.runtime.extension.builtin;
+
+import org.spockframework.runtime.InvalidSpecException;
+import org.spockframework.runtime.extension.AbstractAnnotationDrivenExtension;
+import org.spockframework.runtime.extension.AbstractGlobalExtension;
+import org.spockframework.runtime.model.FeatureInfo;
+import org.spockframework.runtime.model.SpecInfo;
+import spock.lang.Rollup;
+import spock.lang.Unroll;
+
+public class RollupExtension extends AbstractAnnotationDrivenExtension<Rollup> {
+  @Override
+  public void visitSpecAnnotation(Rollup rollup, SpecInfo spec) {
+    visitSpecAnnotation(rollup, spec, true);
+  }
+
+  private void visitSpecAnnotation(Rollup rollup, SpecInfo spec, boolean direct) {
+    if (direct && (spec.getAnnotation(Unroll.class) != null)) {
+      throw new InvalidSpecException("@Unroll and @Rollup must not be used on the same element");
+    }
+
+    for (FeatureInfo feature : spec.getFeatures()) {
+      if (feature.isParameterized()) {
+        visitFeatureAnnotation(rollup, feature, false);
+      }
+    }
+  }
+
+  @Override
+  public void visitFeatureAnnotation(Rollup rollup, FeatureInfo feature) {
+    visitFeatureAnnotation(rollup, feature, true);
+  }
+
+  private void visitFeatureAnnotation(Rollup rollup, FeatureInfo feature, boolean direct) {
+    if (direct && (feature.getFeatureMethod().getAnnotation(Unroll.class) != null)) {
+      throw new InvalidSpecException("@Unroll and @Rollup must not be used on the same element");
+    }
+
+    if (!feature.isParameterized()) return; // could also throw exception if direct
+
+    feature.setReportIterations(false);
+    feature.setIterationNameProvider(null);
+  }
+
+  public static class AutoApply extends AbstractGlobalExtension {
+    private final boolean globalRollup = Boolean.getBoolean("spock.globalRollup");
+    private final Rollup rollup = Rolled.class.getAnnotation(Rollup.class);
```

Comment:
I would just omit the annotation from the private method signature, as it isn't used anyway.

Reply:
Ah, good point, even for the `Unroll` it is not really necessary.